File: solutions/03 annoying blobs/java/src/eu/sioux/tdd/blob/DataAdder.java
package eu.sioux.tdd.blob;

public class DataAdder {
    private DataBlob dataBlob;
    private int newData = 0;

    public DataAdder(DataBlob dataBlob) {
        this.dataBlob = dataBlob;
    }

    public void doSomeAdding() {
        dataBlob.addData(newData);
        newData += 10;
    }

}

File: exercises/03 annoying blobs/cpp/AnnoyingBlobs/AnnoyingBlobs/DataAdder.cpp
#include "DataAdder.h"
#include "DataBlob.h"

DataAdder::DataAdder() :
	_newData(0)
{
}


DataAdder::~DataAdder()
{
}

void DataAdder::doSomeAdding()
{
	DataBlob::getBlob().addData(_newData);
	_newData += 10;
}
